From e0a2b74fdec3bdaaf9d3ddf5673c1e5276f845a7 Mon Sep 17 00:00:00 2001
From: Roman Gushchin <guro@fb.com>
Date: Fri, 25 Nov 2022 11:56:12 +0800
Subject: [PATCH] bpftool: recognize scheduler programs
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e0a2b74fdec3bdaaf9d3ddf5673c1e5276f845a7
Modified-by-SEL: Yes, heavily modified due to different context, need
		 review carefully.


maillist inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5F6X6
CVE: NA

Reference: https://lore.kernel.org/all/20210916162451.709260-1-guro@fb.com/

-------------------

Teach bpftool to recognize scheduler bpf programs.

Signed-off-by: Roman Gushchin <guro@fb.com>
Signed-off-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Ren Zhijie <renzhijie2@huawei.com>
Signed-off-by: Hui Tang <tanghui20@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 tools/bpf/bpftool/common.c |    1 +
 tools/bpf/bpftool/prog.c   |    1 +
 tools/lib/bpf/bpf.c        |    1 +
 3 files changed, 3 insertions(+)

--- a/tools/bpf/bpftool/common.c
+++ b/tools/bpf/bpftool/common.c
@@ -75,6 +75,7 @@ const char * const attach_type_name[__MA
 	[BPF_SK_REUSEPORT_SELECT]	= "sk_skb_reuseport_select",
 	[BPF_SK_REUSEPORT_SELECT_OR_MIGRATE]	= "sk_skb_reuseport_select_or_migrate",
 	[BPF_PERF_EVENT]		= "perf_event",
+	[BPF_SCHED]			= "sched",
 };
 
 void p_err(const char *fmt, ...)
--- a/tools/bpf/bpftool/prog.c
+++ b/tools/bpf/bpftool/prog.c
@@ -67,6 +67,7 @@ const char * const prog_type_name[] = {
 	[BPF_PROG_TYPE_EXT]			= "ext",
 	[BPF_PROG_TYPE_LSM]			= "lsm",
 	[BPF_PROG_TYPE_SK_LOOKUP]		= "sk_lookup",
+	[BPF_PROG_TYPE_SCHED]			= "sched",
 };
 
 const size_t prog_type_name_size = ARRAY_SIZE(prog_type_name);
--- a/tools/lib/bpf/bpf.c
+++ b/tools/lib/bpf/bpf.c
@@ -483,6 +483,7 @@ static int bpf_load_program_xattr2(const
 		break;
 	case BPF_PROG_TYPE_TRACING:
 	case BPF_PROG_TYPE_EXT:
+	case BPF_PROG_TYPE_SCHED:
 		p.attach_btf_id = load_attr->attach_btf_id;
 		p.attach_prog_fd = load_attr->attach_prog_fd;
 		break;
